src/lib.rs

```diff
@@ -14,15 +14,18 @@ use unicode_normalization::UnicodeNormalization;
 
 pub mod tables;
 
+/// Describes why a string failed stringprep normalization.
 #[derive(Debug)]
-enum ErrorCause {
+pub enum ErrorCause {
```

Comment:
That does make sense, but you expose `Error`:

https://github.com/sfackler/rust-stringprep/blob/master/src/lib.rs#L25

And keeping `ErrorCause` private gives no way to match it:

```
 --> tests/saslprep_tests.rs:8:13
  |
8 |         Err(Error(_)) => (),
  |             ^^^^^
  |             |
  |             did you mean `Error { /* fields */ }`?
  |             constructor is not visible here due to private fields
```

Your call, of course, but I think it needs to be fixed in one direction or the other.